### ISSUE TITLE
Build system doesn't prefer internal includes over external

### ISSUE THREAD

**Issue by jekstrand** [KhronosGroup/Vulkan-ValidationLayers#1103]
I ran into this issue this week while trying to rev my build of the validation layers.  If you have an old version of the validation layers installed and one of the include directories pulled in for the loader or other components, the build system adds include paths in the wrong order and may accidentally pull generated headers from the old installed version rather than the newly generated version.  All that needs to be done is to massage the build system so that it includes the local generated headers directory first and any system or headers from `find_package()` afterwords.  I tried to resolve this and post a PR but I quickly ran out of CMake knowledge to try and sort out all the include issues.

**Comment by Tony-LunarG** [comment]
When you say "validation layers installed", do you mean via the SDK, or through a "make install"?  Is this on Windows or Linux or both?  Could you give an example of a generated file that is getting included from the wrong location?

**Comment by jekstrand** [comment]
The setup where I encountered this issue is one in which I'd manually built and installed a bunch of stuff (Vulkan-Headers, SPIRV-Headers, SPIRV-Tools, GLSLang, Vulkan-ValidationLayers) to $HOME/khronos/_install on my Linux machine.  The builds were all valid but were about 3 months old.  I then went to update things one after another.  When I came to the validation layers, they were picking up $HOME/khronos/_install/include on the include path because of Vulkan-Headers (I think).  Then the old installed versions of the Vulkan-ValidationLayers generated headers were clashing with the versions in my updated Vulkan-ValidationLayers repo.

I suspect the same thing would happen if you tried to build against an older SDK but I've not actually tried it personally.
